app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

```diff
@@ -721,11 +721,69 @@ class ExplorationActivityTest {
       internalProfileId,
       topicId,
       storyId,
-      explorationId, /* backflowScreen= */
-      null
+      explorationId,
+      /* backflowScreen= */ null
     )
   }
 
+  private fun setupAudio() {
+    // Only initialize the Robolectric shadows when running on Robolectric (and use reflection since
+    // Espresso can't load Robolectric into its classpath).
+    if (isOnRobolectric()) {
+      val dataSource = createAudioDataSource(
+        explorationId = RATIOS_EXPLORATION_ID_0, audioFileName = "content-en-057j51i2es.mp3"
+      )
+      addShadowMediaPlayerException(dataSource, IOException("Test does not have networking"))
+    }
+  }
+
+  private fun setupAudioForFraction() {
+    // Only initialize the Robolectric shadows when running on Robolectric (and use reflection since
+    // Espresso can't load Robolectric into its classpath).
+    if (isOnRobolectric()) {
+      val dataSource = createAudioDataSource(
+        explorationId = FRACTIONS_EXPLORATION_ID_0, audioFileName = "content-en-nb3k4zuyir.mp3"
+      )
+      val dataSource2 = createAudioDataSource(
+        explorationId = FRACTIONS_EXPLORATION_ID_0, audioFileName = "content-hi-en-l8ik9pdxj2a.mp3"
+      )
+      addShadowMediaPlayerException(dataSource, IOException("Test does not have networking"))
+      addShadowMediaPlayerException(dataSource2, IOException("Test does not have networking"))
+    }
+  }
+
+  private fun addShadowMediaPlayerException(dataSource: Any, exception: Exception) {
+    val classLoader = StateFragmentTest::class.java.classLoader!!
+    val shadowMediaPlayerClass = classLoader.loadClass("org.robolectric.shadows.ShadowMediaPlayer")
+    val addException =
+      shadowMediaPlayerClass.getDeclaredMethod(
+        "addException", dataSource.javaClass, IOException::class.java
+      )
+    addException.invoke(/* obj= */ null, dataSource, exception)
+  }
+
+  private fun isOnRobolectric(): Boolean {
+    return ApplicationProvider.getApplicationContext<ExplorationActivityTest.TestApplication>()
+      .isOnRobolectric()
+  }
+
+  @Suppress("SameParameterValue")
+  private fun createAudioDataSource(explorationId: String, audioFileName: String): Any {
+    val audioUrl = createAudioUrl(explorationId, audioFileName)
+    val classLoader = StateFragmentTest::class.java.classLoader!!
+    val dataSourceClass = classLoader.loadClass("org.robolectric.shadows.util.DataSource")
+    val toDataSource =
+      dataSourceClass.getDeclaredMethod(
+        "toDataSource", String::class.java, Map::class.java
+      )
+    return toDataSource.invoke(/* obj= */ null, audioUrl, /* headers= */ null)
```

Comment:
There is warning in this line have a look.

Reply:
Done

Reply:
Use named arguments instead ?

Reply:
Here and elsewhere too

Reply:
It seems like, named argument is not supported here.